cylc/flow/job_pool.py

```diff
@@ -109,8 +109,8 @@ def remove_job(self, job_d):
     def remove_task_jobs(self, task_id):
         """removed all jobs associated with a task from the pool."""
         name, point_string = TaskID.split(task_id)
-        t_id = f"/{point_string}/{name}/"
-        for job_d in self.pool.keys():
+        t_id = f'{point_string}/{name}/'
+        for job_d in list(self.pool.keys()):
```

Comment:
No need to call `.keys()`?

```suggestion
        for job_d in list(self.pool):
```

Reply:
Fixed.